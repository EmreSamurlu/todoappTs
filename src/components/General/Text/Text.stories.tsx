import Text, {type TxtProps} from './Text';

import type {Meta, StoryObj} from '@storybook/react-native';

const TextMeta: Meta<TxtProps> = {
  args: {
    text: 'Hello World',
  },
  title: 'Text',
  component: Text,
};

export default TextMeta;

export const Basic: StoryObj<TxtProps> = {};
