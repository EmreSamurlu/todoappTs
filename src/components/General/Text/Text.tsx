import React from 'react';
import {Text as NativeText, type TextProps} from 'react-native';

import {useTranslation} from 'react-i18next';

export interface TxtProps extends TextProps {
  text: string;
  textStyle: object;
}

const Text: React.FC<TxtProps> = ({text, textStyle}) => {
  const {t} = useTranslation();
  return <NativeText style={textStyle}>{t(text)}</NativeText>;
};

export default Text;
