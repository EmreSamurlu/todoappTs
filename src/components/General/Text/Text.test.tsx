import {describe, expect, jest, test} from '@jest/globals';
import React from 'react';
import {render} from '@testing-library/react-native';
import Text from './Text';

jest.mock('react-i18next', () => ({
  // this mock makes sure any components using the translate hook can use it without a warning being shown
  useTranslation: () => {
    return {
      t: (str: string) => str,
      i18n: {
        changeLanguage: () => new Promise(() => {}),
        // You can include here any property your component may use
      },
    };
  },
}));

describe('describe', () => {
  test('test', () => {
    const component = render(
      <Text
        text="Hello world"
        textStyle={{
          fontSize: 13,
        }}
      />,
    );
    expect(component).toMatchSnapshot();
  });
});
