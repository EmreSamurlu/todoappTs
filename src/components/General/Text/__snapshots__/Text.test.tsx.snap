// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `<Text />`;

exports[`describe test 1`] = `
<Text
  style={
    {
      "fontSize": 13,
    }
  }
>
  Hello world
</Text>
`;
